tests/Mixins/DownloadCollectionTest.php

```diff
@@ -26,4 +27,52 @@ public function can_store_a_collection_as_excel()
             $response->headers->get('Content-Disposition')
         );
     }
+
+    /**
+     * @test
+     */
+    public function can_store_a_collection_with_headers_as_excel()
+    {
+        $collection = new Collection([
+            ['column_1' => 'test', 'column_2' => 'test'],
+            ['column_1' => 'test', 'column_2' => 'test'],
+        ]);
+
+        $response = $collection->downloadExcel('collection-headers-download.xlsx', Excel::XLSX, true);
+
+        $array = $this->readAsArray($response->getFile()->getPathName(), Excel::XLSX);
+
+        $this->assertEquals(['column_1', 'column_2'], collect($array)->first());
+        $this->assertInstanceOf(BinaryFileResponse::class, $response);
+        $this->assertEquals(
+            'attachment; filename="collection-headers-download.xlsx"',
+            $response->headers->get('Content-Disposition')
+        );
+    }
+
+    /**
+     * @test
+     */
+    public function can_store_a_collection_without_headers_as_excel()
+    {
+        $collection = new Collection([
+            ['column_1' => 'test', 'column_2' => 'test'],
+            ['column_1' => 'test', 'column_2' => 'test'],
+        ]);
+
+        $response = $collection->downloadExcel('collection-without-headers-download.xlsx', Excel::XLSX, false);
+
+        $array = $this->readAsArray($response->getFile()->getPathName(), Excel::XLSX);
+
+        $firstRow = collect($array)->first();
+        $this->assertNotEquals(['column_1', 'column_2'], $firstRow);
+        $this->assertNotEquals([], $firstRow);
+        $this->assertNotNull($firstRow);
```

Comment:
Can you add an assert here to check what it does equals to?

Reply:
I have replaced by an assertEquals and changed the data so each row is unique, it's better i think.